dashboard/src/types/hardware/hardwareDetails.ts

```diff
@@ -94,7 +95,75 @@ export const zDiffFilter = z
       bootDurationMax: zFilterNumberValue,
       testDurationMin: zFilterNumberValue,
       testDurationMax: zFilterNumberValue,
+      trees: zFilterBoolValue,
     } satisfies Record<TFilterKeys, unknown>),
     z.record(z.never()),
   ])
   .catch({});
+
+export type TFilter = z.infer<typeof zDiffFilter>;
+export type TFilterObjectsKeys = z.infer<typeof zFilterObjectsKeys>;
+export type TFilterNumberKeys = z.infer<typeof zFilterNumberKeys>;
+
+export const isTFilterObjectKeys = (key: string): key is TFilterObjectsKeys => {
+  return zFilterObjectsKeys.safeParse(key).success;
+};
+
+export const isTFilterNumberKeys = (key: string): key is TFilterNumberKeys => {
+  return zFilterNumberKeys.safeParse(key).success;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
add TODO to remove this

Reply:
just change the declartion to a type
```suggestion
type requestFilters = {
```

Reply:
I use `const targetFilter: readonly string[] = requestFilters[target];` in a function below. Change for type give a error in typescript

Reply:
If you are using it you don't need the eslint-disable